### PR TITLE
Bcd to tzkt migration

### PR DESCRIPTION
**Pull request by letier3110**
https://www.notion.so/madfissolutions/BCD-API-users-to-migrate-to-the-new-endpoints-6a514aedca6d4637bc49e2b13dd16337

### CODE REVIEW

File: src/interfaces/activity.interface.mock.ts [hunk 1]
```diff
@@ -50,54 +51,85 @@ const mockAppliedDelegationOperation: ActivityInterface = {
...
-  entrypoint: 'transfer',
-  source: {
-    address: 'address4',
-    alias: 'alias4'
+// const mockPendingTransfer: ActivityInterface = {
+//   type: ActivityTypeEnum.Transaction,
+//   status: ActivityStatusEnum.Pending,
+//   hash: 'hash2',
+//   amount: '1',
+//   timestamp: 9,
```
Comment (LendiHop) [inline src/interfaces/activity.interface.mock.ts:76]: delete

File: src/interfaces/activity.interface.mock.ts [hunk 2]
```diff
@@ -50,54 +51,85 @@ const mockAppliedDelegationOperation: ActivityInterface = {
...
 
-export const mockOperations: ActivityInterface[] = [
+// const mockAppliedTransfer: ActivityInterface = {
+//   type: ActivityTypeEnum.Transaction,
+//   status: ActivityStatusEnum.Applied,
+//   hash: 'hash4',
+//   amount: '0',
+//   address: 'contractAddress1',
+//   id: 0,
+//   timestamp: 8,
```
Comment (LendiHop) [inline src/interfaces/activity.interface.mock.ts:109]: delete

**Review by LendiHop — APPROVED** [review]

File: src/interfaces/tzkt.interface.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+export interface TzktAlias {
```
Comment (unixvb) [inline src/interfaces/tzkt.interface.ts:1]: do not export

File: src/interfaces/activity.interface.mock.ts [hunk 4]
```diff
@@ -50,54 +51,55 @@ const mockAppliedDelegationOperation: ActivityInterface = {
...
 
-export const mockAppliedActivityGroups: ActivityGroup[] = groupActivitiesByHash(
+export const mockAppliedActivityGroups: Array<ActivityGroup> = groupActivitiesByHash(
   [mockAppliedOperation, mockAppliedDelegationOperation],
-  [mockAppliedTransfer]
-);
-export const mockPendingActivityGroups: ActivityGroup[] = groupActivitiesByHash(
-  [mockPendingOperation],
-  [mockPendingTransfer]
+  []
```
Comment (unixvb) [inline src/interfaces/activity.interface.mock.ts:98]: why this is empty?

File: src/utils/operation.utils.ts [hunk 5]
```diff
@@ -15,59 +15,56 @@ export const mapOperationsToActivities = (address: string, operations: Operation
...
-          console.log(`Ignoring kind ${type}`);
+      case ActivityTypeEnum.Origination:
+        isDefined(originatedContract) && (destination = originatedContract);
+        isDefined(contractBalance) && (amount = contractBalance.toString());
+        break;
 
-          continue;
-      }
+      default:
+        console.log(`Ignoring kind ${type}`);
```
Comment (unixvb) [inline src/utils/operation.utils.ts:53]: do you need this?

File: src/store/wallet/wallet-epics.ts [hunk 6]
```diff
@@ -178,16 +177,22 @@ const loadActivityGroupsEpic = (action$: Observable<Action>, state$: Observable<
     switchMap(([, selectedAccount]) =>
       forkJoin([
         from(
-          tzktApi.get<OperationInterface[]>(
-            `accounts/${selectedAccount.publicKeyHash}/operations?limit=100&type=${ActivityTypeEnum.Delegation},${ActivityTypeEnum.Origination},${ActivityTypeEnum.Transaction}`
+          tzktApi.get<Array<OperationInterface>>(
+            `accounts/${selectedAccount.publicKeyHash}/operations?limit=1000&type=${ActivityTypeEnum.Delegation},${ActivityTypeEnum.Origination},${ActivityTypeEnum.Transaction}`
```
Comment (unixvb) [inline src/store/wallet/wallet-epics.ts:181]: extract this into fn, same as `getTokenBalances`

File: src/store/wallet/wallet-epics.ts [hunk 7]
```diff
@@ -178,16 +177,22 @@ const loadActivityGroupsEpic = (action$: Observable<Action>, state$: Observable<
...
           )
         ).pipe(map(({ data }) => mapOperationsToActivities(selectedAccount.publicKeyHash, data))),
         from(
-          betterCallDevApi.get<GetAccountTokenTransfersResponseInterface>(
-            `/tokens/${CURRENT_NETWORK_ID}/transfers/${selectedAccount.publicKeyHash}`,
-            { params: { max: 100, start: 0 } }
+          tzktApi.get<Array<TzktTokenTransfer>>(
+            `/tokens/transfers?anyof.from.to=${selectedAccount.publicKeyHash}&limit=10000`
```
Comment (unixvb) [inline src/store/wallet/wallet-epics.ts:186]: extract this into fn, same as `getTokenBalances`

File: src/utils/token-balance.utils.ts [hunk 8]
```diff
@@ -3,41 +3,25 @@ import { BigNumber } from 'bignumber.js';
...
-        ).pipe(
-          map(restResponses => [
-            ...initialResponse.data.balances,
-            ...restResponses.map(restResponse => restResponse.data.balances).flat()
-          ])
-        );
-      } else {
-        return of(initialResponse.data.balances);
-      }
+      return of(initialResponse.data);
```
Comment (unixvb) [inline src/utils/token-balance.utils.ts:24]: just do the `map` operator (without `of`)

**Review by unixvb — APPROVED** [review]

**Pull request merged**